src/libs/migrations/KeyReportActionsByReportActionID.js

```diff
@@ -0,0 +1,46 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import Log from '../Log';
+import ONYXKEYS from '../../ONYXKEYS';
+
+/**
+ * This migration updates reportActions data to be keyed by reportActionID rather than by sequenceNumber.
+ *
+ * @returns {Promise}
+ */
+export default function () {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+            waitForCollectionCallback: true,
+            callback: (allReportActions) => {
+                Onyx.disconnect(connectionID);
+                const newReportActions = {};
+
+                for (let i = 0; i < Object.entries(allReportActions).length; i++) {
```

Comment:
```suggestion
                for (let i = 0; i < Object.entries(allReportActions || {}).length; i++) {
```
I'm pretty sure allReport actions can easily be null and then Object.entries can throw an error on null and crash the app so having `|| {}` would have it be safer